src/core/dynamic_batch_scheduler.cc

```diff
@@ -485,6 +485,9 @@ DynamicBatchScheduler::GetDynamicBatch()
   // If we found a preferred batch size and the queue delay hasn't been
   // exceeded, then execute that.
   if ((best_preferred_batch_size != 0) && !delay_is_exceeded) {
+    if (pending_batch_delay_ns_ == 0) {
+      payload_saturated_ = true;
```

Comment:
I think you can just move this to line 511 where it returns early if `pending_batch_delay_ns_ == 0`? Seems to be the same checking

Reply:
This handles the case where preferred_batch_size_ is used with max_queue_delay_microseconds=0. With max_queue_delay_microseconds==0, delay is never exceeded.